### PR TITLE
update helm chart links

### PR DESCRIPTION
**Pull request by radTuti**
## Description

Based on new docs updates + fixed some broken links

## Related issues/PRs

- #7300 

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: hack/postrelease/test_helm_artifacts.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def test_calico_release_in_helm_index():
         % (RELEASE_VERSION, RELEASE_VERSION)
     )
 
-    req = requests.get("https://projectcalico.docs.tigera.io/charts/index.yaml")
```
Comment (caseydavenport) [inline hack/postrelease/test_helm_artifacts.py:23]: We should still expect the old URL to work - we have redirects set up and should have a test to make sure that they continue to work.

Comment (radTuti) [reply hack/postrelease/test_helm_artifacts.py:23]: i'll revert

**Pull request merged**